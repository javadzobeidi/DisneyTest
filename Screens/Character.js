/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, { useState, useEffect } from 'react';
import Icon from 'react-native-vector-icons/AntDesign';
import {
  ScrollView,
  Image,
  View,
} from 'react-native';

import tw from '../Lib/tailwind';
import { Text ,Button} from '../Component'
import { useNavigation } from '@react-navigation/native';


export function Character({ props }) {
  const navigation = useNavigation();
console.log(props);
  return (
    <View style={tw`flex-1`}>
     
        <ScrollView style={tw` bg-background_primary mx-4 my-4`}>
      <View style={tw` mr-4 flex-col   mx-2 my-6 flex-row text-center  items-start  	rounded-lg mb-4 	py-6 	 `} >
        <View style={tw`flex-2 px-3    `}>
          <View style={tw`flex-row py-2 text-justify  flex-1 	`}>
            <Text className=' flex-1 font-bold'>Name:</Text>
            <Text className=' flex-1'>{props.name}</Text>
          </View>

          <View style={tw`flex-row py-2 text-justify  flex-1 	`}>
            <Text className=' flex-1 font-bold'>Films:</Text>
            <Text className='flex-1'>{props.films}</Text>
          </View>

          <View style={tw`flex-row py-2 text-justify  flex-1 	`}>
            <Text className=' flex-1 font-bold'>Short Films :</Text>
            <Text className='flex-1'>{props.shortFilms}</Text>
          </View>

          <View style={tw`flex-row py-2 text-justify  flex-1 	`}>
            <Text className=' flex-1 font-bold'>Video Games :</Text>
            <Text className='flex-1'>{props.videoGames}</Text>
          </View>

 
        </View>

        <View style={tw`flex-1 py-4  items-start	justify-start	`}>
          <View style={tw`  rounded-full bg-indigo-50`}>
            <Image
              style={tw`w-28 h-28 rounded-full`}
              source={{uri:props.imageUrl}}

            />
          </View>

        </View>

      </View>
      
      </ScrollView>
     

      <View style={tw`w-34  px-4 py-4  `}>
      
<Button className='flex flex-row 	' onPress={()=>navigation.navigate('characterList')} title='Back' >
<Icon name="arrowleft" size={20} color={'white'} style={tw`mx-4`}/>
  </Button>
  
</View>

     
      </View>

  );
};

