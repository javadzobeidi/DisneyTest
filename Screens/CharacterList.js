/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

 import React,{useState,useEffect} from 'react';
 import { useQuery } from 'react-query';
 import Icon from 'react-native-vector-icons/AntDesign';

 import {
   ScrollView,
  
   Image,
   FlatList,
   Text,
   Alert ,
   View,
 } from 'react-native';
 import {Button,LoadingLayout} from '../Component'
 import tw from '../Lib/tailwind';
 import {GetCharacterList} from '../Api'
 
 export const CharacterList = ({ navigation }) => {

    const [page,setPage]=useState(1);
    const [response,setReponse]=useState([]);
    const [error,setError]=useState(null);

 const {  isLoading, isSuccess ,data,
    isFetching,
    isRefetching,
    isPreviousData,
} = useQuery(['posts',page], ()=>GetCharacterList(page),{onSuccess ,cacheTime: 0 });





function onSuccess(result) {
  if (result.httpCode==200)
  {
    setReponse(result.response)
  }
  else
  {
    Alert.alert(
      result.message
   );

    //setError(result.message);
  }

}



   const renderItem = ({ item }) => (
     <CharacterListItem data={item} />
   );
 
     return (
        <View>
{(isLoading ||isRefetching)  && (
                   <LoadingLayout />
            )}

          {error && <View stlye={tw`flex`}><Text>{error}</Text>
          

          </View>}

{response &&  (
        <ScrollView  nestedScrollEnabled={true} style={tw`my-5 pl-4    `}>
       <Text style={tw`text-2xl text-black font-bold my-5 `}>List of Characters</Text>
      <View style={tw`w-full`} >
  
      <FlatList
      inverted
        horizontal={false}
      columnWrapperStyle={{flex:1, justifyContent: 'space-between'}}
   data={response.data}
   numColumns={2}
   keyExtractor={(_item, index) => index.toString()}
   renderItem={({ item }) => (
     <CharacterListItem
       data={item} navigation={navigation}
     />
   )}
 />
 <View style={tw`flex-1 flex-row pr-4`}>

  
 <Button className='flex-1 flex flex-row  justify-around  mr-4 h-16  ' disabled={response.previousPage=== undefined} onPress={()=>setPage(page-1)}  >
 <Icon name="arrowleft" size={25} color={'white'} style={tw``}/>
 <Text style={tw`font-bold text-lg	  text-white`}>Back</Text>
   </Button>

     <Button className='flex-1 flex  flex-row justify-around	    h-16' disabled={!response.nextPage} onPress={()=>setPage(page+1)}  >
 <Text style={tw`font-bold text-lg	  text-white`}>Next</Text>
 <Icon name="arrowright" size={25} color={'white'} style={tw``}/>
   </Button> 


 </View>
      </View>
      </ScrollView>
)}
      </View>
        
     );
 };


  const CharacterListItem = ({
    navigation,
    data
  }) => (
   <View style={tw`mr-4 flex-1  text-center items-center justify-center 	rounded-lg mb-4 	py-2 	 bg-background_primary ` } >
   <View style={tw`w-20 h-20 rounded-full bg-indigo-50`}>
       <Image
         style={tw`w-full h-full rounded-full`}
         source={{uri:data.imageUrl}}
         />
     </View>
          <Text style={tw`text-white pt-4`}>{data.name}</Text>
          <View style={tw`w-full px-4 pt-4`}>

     <Button onPress={()=>navigation.navigate('character',{id:data._id})} title='Show More' />
     </View>
        </View>
  )
 
 
  const RightArrow=()=>{
    return (
     <svg class="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M17 8l4 4m0 0l-4 4m4-4H3"></path></svg>
    )
  }
 
 