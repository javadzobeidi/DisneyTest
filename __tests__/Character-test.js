import React from 'react';
import renderer from 'react-test-renderer';
import {Character} from '../Screens/Character'
import { QueryClient, QueryClientProvider } from 'react-query';
const queryClient = new QueryClient();

jest.useFakeTimers();
const mockedNavigate = jest.fn();

const mockedProps=`{
  "_id": 6,
  "films": [
    "Alice in Wonderland (2010 film)",
    "Alice Through the Looking Glass"
  ],
  "shortFilms": [
    "Alice in Wonderland (2010 film)",
    "Alice Through the Looking Glass"
  ],
  "tvShows": [
    "Alice in Wonderland (2010 film)",
    "Alice Through the Looking Glass"
  ],
  "videoGames": [
    "Alice in Wonderland (2010 film)",
    "Alice Through the Looking Glass"
  ],
  "parkAttractions": [
    
  ],
  "allies": [
    
  ],
  "enemies": [
    
  ],
  "name": "'Olu Mel",
  "imageUrl": "https://static.wikia.nocookie.net/disney/images/6/61/Olu_main.png",
  "url": "https://api.disneyapi.dev/characters/6"
}`



jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

test('renders correctly', () => {
  const tree = renderer.create(
    <QueryClientProvider client={queryClient}>
    <Character props={mockedProps} />
</QueryClientProvider>
).toJSON();
  expect(tree).toMatchSnapshot();
});
