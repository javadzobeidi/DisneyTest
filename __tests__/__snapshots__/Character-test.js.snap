// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#3A3F47",
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 16,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "borderRadius": 8,
            "flexDirection": "row",
            "marginBottom": 16,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 24,
            "paddingBottom": 24,
            "paddingTop": 24,
            "textAlign": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flexBasis": "0%",
              "flexGrow": 2,
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          <View
            style={
              Object {
                "flexBasis": "0%",
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingBottom": 8,
                "paddingTop": 8,
                "textAlign": "justify",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontWeight": "bold",
                }
              }
            >
              Name:
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flexBasis": "0%",
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingBottom": 8,
                "paddingTop": 8,
                "textAlign": "justify",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontWeight": "bold",
                }
              }
            >
              Films:
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flexBasis": "0%",
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingBottom": 8,
                "paddingTop": 8,
                "textAlign": "justify",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontWeight": "bold",
                }
              }
            >
              Short Films :
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flexBasis": "0%",
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingBottom": 8,
                "paddingTop": 8,
                "textAlign": "justify",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontWeight": "bold",
                }
              }
            >
              Video Games :
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "paddingBottom": 16,
              "paddingTop": 16,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#eef2ff",
                "borderRadius": 9999,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "borderRadius": 9999,
                  "height": 112,
                  "width": 112,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": 136,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": "",
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#8B9A47",
          "borderRadius": 8,
          "display": "flex",
          "elevation": 3,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.125,
          "shadowRadius": 3,
          "width": "100%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "marginLeft": 16,
              "marginRight": 16,
            },
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î˜œ
      </Text>
      <Text
        style={
          Object {
            "color": "#fff",
            "textAlign": "center",
          }
        }
      >
        Back
      </Text>
    </View>
  </View>
</View>
`;
